modules/orakiBidAdapter.js

```diff
@@ -0,0 +1,239 @@
+import { logMessage, logError, deepAccess } from '../src/utils.js';
+import { convertOrtbRequestToProprietaryNative } from '../src/native.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import { config } from '../src/config.js';
+
+const BIDDER_CODE = 'oraki';
+const AD_URL = 'https://eu1.oraki.io/pbjs';
+const SYNC_URL = 'https://sync.oraki.io';
+
+function isBidResponseValid(bid) {
+  if (!bid.requestId || !bid.cpm || !bid.creativeId || !bid.ttl || !bid.currency) {
+    return false;
+  }
+
+  switch (bid.mediaType) {
+    case BANNER:
+      return Boolean(bid.width && bid.height && bid.ad);
+    case VIDEO:
+      return Boolean(bid.vastUrl || bid.vastXml);
+    case NATIVE:
+      return Boolean(bid.native && bid.native.impressionTrackers && bid.native.impressionTrackers.length);
+    default:
+      return false;
+  }
+}
+
+function getPlacementReqData(bid) {
```

Comment:
Why do 21 bid adapters use this exact same code block. Is one person committing twenty bid adapters?

If so, can you start to move code blocks into to libraries for better maintenance?